addon/utils/notify-changes.js

```diff
@@ -0,0 +1,53 @@
+import Ember from 'ember';
+import { notifyPropertyChange as _notifyPropertyChange } from '@ember/object';
+import { assert } from '@ember/debug';
+import { DEBUG } from '@glimmer/env';
+
+const { changeProperties } = Ember;
+
+const { propertyDidChange } = Ember;
+export let notifyPropertyChange;
+
+const HasNotifyPropertyChange = _notifyPropertyChange !== undefined;
+if (HasNotifyPropertyChange) {
+  notifyPropertyChange = _notifyPropertyChange;
+} else {
+  notifyPropertyChange = propertyDidChange;
+}
+
+const StoreToChanges = new WeakMap();
+
+export function beginChanges(store) {
+  StoreToChanges.set(store, []);
+}
+
+export function deferPropertyChange(store, obj, key) {
+  if (!StoreToChanges.has(store)) {
+    beginChanges(store);
+  }
+  let changes = StoreToChanges.get(store);
+  changes.push(obj, key);
+}
+
+export function flushChanges(store) {
+  let changes = StoreToChanges.get(store) || [];
+  changeProperties(() => {
+    for (let i = 0; i < changes.length; i += 2) {
+      let obj = changes[i];
+      let key = changes[i + 1];
+      notifyPropertyChange(obj, key);
+    }
+  });
+  StoreToChanges.set(store, []);
+}
+
+export function assertNoChanges(store) {
```

Comment:
I really like this, is there a way to maybe schedule it at the end of a runloop every time we call `defer` so we are always sure to not have stale notifications?

Reply:
that's a great idea.  #269